learning/katas/go/Core Transforms/CoGroupByKey/CoGroupByKey/task.md

```diff
@@ -0,0 +1,104 @@
+    to you under the Apache License, Version 2.0 (the
+    "License"); you may not use this file except in compliance
+    with the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing,
+    software distributed under the License is distributed on an
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+-->
+
+# CoGroupByKey
+
+CoGroupByKey performs a relational join of two or more key/value PCollections that have the same 
+key type.
+
+**Kata:** Implement a [beam.CoGroupByKey](https://godoc.org/github.com/apache/beam/sdks/go/pkg/beam#CoGroupByKey) 
+transform that join words by the first alphabetical letter, and then produces the string representation of the 
+WordsAlphabet model.
+
+<div class="hint">
+    Refer to
+    <a href="https://godoc.org/github.com/apache/beam/sdks/go/pkg/beam#CoGroupByKey">beam.CoGroupByKey</a>
+    to solve this problem.
+</div>
+
+<div class="hint">
+  Refer to the Beam Programming Guide
+  <a href="https://beam.apache.org/documentation/programming-guide/#cogroupbykey">
+    "CoGroupByKey"</a> section for more information.
+</div>
+
+<div class="hint">
+  Think of this problem in three stages.  First, create key/value pairs of PCollections called KV
+  for fruits and countries, pairing the first character with the word.  Next, apply CoGroupByKey to the KVs
+  followed by a ParDo.
+</div>
+
+<div class="hint">
+  In the last lesson we learned how to make key/value PCollections called KV.  Now we have 
+  two to make from fruits and countries.
+  
+  To return as a KV, you can return two values from your DoFn. The first return value represents the Key, and 
+  the second return value represents the Value.  An example is shown below.
+  
+```
+func doFn(element string) (string, string) {
+    key := string(element[0])
+    value := element
+    return key, value
+}
+``` 
+</div>
+
+<div class="hint">
+  In the last lesson we learned that 
+  <a href="https://godoc.org/github.com/apache/beam/sdks/go/pkg/beam#GroupByKey">
+  beam.GroupByKey</a> takes a single KV.
+  <a href="https://godoc.org/github.com/apache/beam/sdks/go/pkg/beam#CoGroupByKey">beam.CoGroupByKey</a>
+  takes more than one KV.
+</div>
+
+<div class="hint">
+  Our final step in this problem requires a
+  <a href="https://godoc.org/github.com/apache/beam/sdks/go/pkg/beam#ParDo">beam.ParDo</a>
+  with a DoFn that's different than what we've seen in previous lessons.  In the previous step we should
+  have a PCollection acquired from CoGroupByKey.  A ParDo for that PCollection expects a DoFn that looks
+  like the following. 
+  
+  ```
+  func doFn(key string, aKV func(*string) bool, anotherKV func(*string) bool, emit func(string)){
```

Comment:
If I understand correctly, it seems that `func(*string) bool` should assign the value of V to the passed-in variable pointer, not the KV?
Borrowing the concept of the Java version, these functions behave similarly to CoGbkResult.get() based on the sequence of the passed in PCollection?

Reply:
Yes I believe this is correct.